homeassistant/components/syncthing/config_flow.py

```diff
@@ -0,0 +1,82 @@
+"""Config flow for syncthing integration."""
+import logging
+
+import aiosyncthing
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_NAME, CONF_TOKEN, CONF_URL, CONF_VERIFY_SSL
+
+from .const import DEFAULT_NAME, DEFAULT_URL, DEFAULT_VERIFY_SSL, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME, default=DEFAULT_NAME): str,
+        vol.Required(CONF_URL, default=DEFAULT_URL): str,
+        vol.Required(CONF_TOKEN): str,
+        vol.Required(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): bool,
+    }
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect."""
+
+    if DOMAIN in hass.data and data[CONF_NAME] in hass.data[DOMAIN]:
+        raise AlreadyConfigured
+
+    try:
+        async with aiosyncthing.Syncthing(
+            data[CONF_TOKEN],
+            url=data[CONF_URL],
+            verify_ssl=data[CONF_VERIFY_SSL],
+            loop=hass.loop,
+        ) as client:
+            await client.system.config()
+            return {"title": f"{data[CONF_NAME]} ({data[CONF_URL]})"}
+    except aiosyncthing.exceptions.UnauthorizedError as error:
+        raise InvalidAuth from error
+    except Exception as error:
+        raise CannotConnect from error
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for syncthing."""
+
+    VERSION = 1
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            try:
+                info = await validate_input(self.hass, user_input)
+
+                return self.async_create_entry(title=info["title"], data=user_input)
```

Comment:
But why? As far as I can see many other integrations follow the same pattern, and this code was generated by the scaffolding

Reply:
Because the try/except clause should only contain the code you are trying to catch error from.
The entry creation should not be part of this, as you are trying the catch error thrown from the validate_input, not the entry creation.